src/Integration.Vsix/SonarLintDaemon/Protocol.cs

```diff
@@ -0,0 +1,231 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace SonarLint.VisualStudio.Integration.Vsix.SonarLintDaemon
+{
+
+    class Request
+    {
+        public const int Verify = 1;
+        public const int C99 = 1 << 1;
+        public const int C11 = 1 << 2;
+        public const int CPlusPlus = 1 << 3;
+        public const int CPlusPlus11 = 1 << 4;
+        public const int CPlusPlus14 = 1 << 5;
+        public const int CPlusPlus17 = 1 << 6;
+        public const int ObjC = 1 << 7;
+        public const int MS = 1 << 8;
+        public const int GNU = 1 << 9;
+        public const int OperatorNames = 1 << 10;
+        public const int CharIsUnsigned = 1 << 11;
+        public const int CreateReproducer = 1 << 14;
+        public const int AutoRefCount = 1 << 15;
+        public const int Weak = 1 << 16;
+        public const int C17 = 1 << 17;
+
+        public string[] Options { get; set; } = Array.Empty<string>();
+        public long Flags { get; set; }
+        public long MsVersion { get; set; }
+        public string[] IncludeDirs { get; set; } = Array.Empty<string>();
+        public string[] FrameworkDirs { get; set; } = Array.Empty<string>();
+        public string[] VfsOverlayFiles { get; set; } = Array.Empty<string>();
+        public string ModuleName { get; set; } = "";
+        public string Predefines { get; set; } = "";
+        public string[] Macros { get; set; } = Array.Empty<string>();
+        public string TargetTriple { get; set; } = "x86_64-unknown-unknown";
+        public string File { get; set; } = "";
+    }
+
+    class Response
+    {
+        public Message[] Messages { get; set; }
+
+        public Response(Message[] messages)
+        {
+            Messages = messages;
+        }
+    }
+
+    class MessagePart
+    {
+        public string Filename;
+        public int Line { get; }
+        public int Column { get; }
+        public int EndLine { get; }
+        public int EndColumn { get; }
+        public string Text { get; }
+
+        public MessagePart(
+      String filename,
```

Comment:
Weird indentation of the parameters.
If you delete then retype the final `}` in the document VS will reformat the indents (same as `Edit`, `Advanced`, `Format document` from the main menu).